### DISCUSSION TITLE
Do we need to use [ScopedService] for services registered using AddScoped in DI?

### DISCUSSION THREAD

**Discussion by gojanpaolo** [ChilliCream/hotchocolate#3117] [Q&A] [answered]
[The graphql-workshop explained the need to use `[ScopedService]` for a `DbContext` because of thread-safety issues](https://github.com/ChilliCream/graphql-workshop/blob/master/docs/3-understanding-dataLoader.md). Do we need to use `[ScopedService]` for services that were added in DI using `AddScoped`? Or is it ok to just use `[Service]`?

I looked at the [ScopedServiceAttribute](https://github.com/ChilliCream/hotchocolate/blob/main/src/HotChocolate/Core/src/Abstractions/ScopedServiceAttribute.cs) and [ServiceAttribute](https://github.com/ChilliCream/hotchocolate/blob/main/src/HotChocolate/Core/src/Abstractions/ServiceAttribute.cs) source code but there's nothing in it that would describes its behavior

Our scoped service implementation basically caches some information derived from the `HttpRequest`. The service uses our `DbContext` as well but it's instantiated manually and not relying on DI for injecting it. The values cached are also lazy loaded. Our scoped service implementation should be thread-safe.

Thanks!

**Comment by michaelstaib** [comment, marked as answer]
The `[ScopedServiceAttribute]` marks resolver scoped services. These are services that are stored on the scoped context data.

if you just want to access a standard scoped server from the DI either use `[Service]` or `[FromServices]`.

**Reply by gojanpaolo** [reply]
Got it! Thank you!
